server/fleet/datastore.go

```diff
@@ -975,6 +976,10 @@ func IsNotFound(err error) bool {
 	if errors.As(err, &nfe) {
 		return nfe.IsNotFound()
 	}
+	// consider sql.ErrNoRows as not found too
+	if errors.Is(err, sql.ErrNoRows) {
+		return true
+	}
```

Comment:
I'm probably being too conservative, but this scares me a little, I wonder if for any endpoints we're relying on `sql.ErrNoRows` to be any other kind of error.

Reply:
Yeah, my reasoning here was the tests pass and if we forget to wrap `sql.ErrNotFound` to a `notFound` error in the datastore layer, it would still be properly detected as "not found". If you have second thoughts, I'm happy to remove it, I'll just need to update a couple of places in tests that use the mock datastore that return `sql.ErrNoRows` instead of an error that actually implements `fleet.NotFoundError`.

Reply:
Reverted that and used a `notFoundError` in the test instead.